diff --git a/.config/dotnet-tools.json b/.config/dotnet-tools.json
index 7a7c21b..6b99e20 100644
--- a/.config/dotnet-tools.json
+++ b/.config/dotnet-tools.json
@@ -4,17 +4,18 @@
   "tools": {
     "hotchocolate.fusion.commandline": {
       "version": "16.0.0-p.7.17",
-      "commands": [
-        "fusion"
-      ],
+      "commands": ["fusion"],
+      "rollForward": false
+    },
+    "chillicream.nitro.commandline": {
+      "version": "16.0.0-p.7.35",
+      "commands": ["nitro"],
       "rollForward": false
     },
     "aspire.cli": {
       "version": "9.4.1",
-      "commands": [
-        "aspire"
-      ],
+      "commands": ["aspire"],
       "rollForward": false
     }
   }
-}
\ No newline at end of file
+}
diff --git a/Directory.Packages.props b/Directory.Packages.props
index 30f34d3..8284f3c 100644
--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -3,23 +3,27 @@
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
   </PropertyGroup>
   <ItemGroup Label="HotChocolate">
-    <PackageVersion Include="HotChocolate" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.AspNetCore" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.AspNetCore.Authorization" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.AspNetCore.CommandLine" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Data.EntityFramework" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Fusion.Composition" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Fusion.Packaging" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Subscriptions.Redis" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Fusion.AspNetCore" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Fusion.Aspire" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Diagnostics" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Execution" Version="16.0.0-p.7.31" />
-    <PackageVersion Include="HotChocolate.Types.Analyzers" Version="16.0.0-p.7.31">
+    <PackageVersion Include="ChilliCream.Nitro" Version="16.0.0-preview.6" />
+    <PackageVersion Include="HotChocolate" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.AspNetCore" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.AspNetCore.Authorization" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.AspNetCore.CommandLine" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Data.EntityFramework" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Fusion.Composition" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Fusion.Packaging" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Subscriptions.Redis" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Fusion.AspNetCore" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Fusion.Aspire" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Diagnostics" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Execution" Version="16.0.0-p.7.35" />
+    <PackageVersion Include="HotChocolate.Types.Analyzers" Version="16.0.0-p.7.35">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageVersion>
   </ItemGroup>
+  <ItemGroup Label="ChilliCream">
+    <PackageVersion Include="ChilliCream.Nitro.Fusion" Version="16.0.0-preview.6" />
+  </ItemGroup>
   <ItemGroup Label="Aspire">
   </ItemGroup>
   <ItemGroup Label="Dependencies">
diff --git a/src/Defaults/ServiceDefaults/Extensions.cs b/src/Defaults/ServiceDefaults/Extensions.cs
index e17553a..15394fe 100644
--- a/src/Defaults/ServiceDefaults/Extensions.cs
+++ b/src/Defaults/ServiceDefaults/Extensions.cs
@@ -43,7 +43,6 @@ public static class Extensions
         string name,
         string version)
     {
-        builder.Logging.ClearProviders();
         builder.Logging.AddOpenTelemetry(logging =>
         {
             logging.IncludeFormattedMessage = true;
diff --git a/src/Defaults/SourceSchemaDefaults/Demo.SourceSchemaDefaults.csproj b/src/Defaults/SourceSchemaDefaults/Demo.SourceSchemaDefaults.csproj
index 60a77f4..afd590b 100644
--- a/src/Defaults/SourceSchemaDefaults/Demo.SourceSchemaDefaults.csproj
+++ b/src/Defaults/SourceSchemaDefaults/Demo.SourceSchemaDefaults.csproj
@@ -5,8 +5,9 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="HotChocolate.Diagnostics"/>
-        <PackageReference Include="HotChocolate.Data.EntityFramework"/>
+        <PackageReference Include="ChilliCream.Nitro" />
+        <PackageReference Include="HotChocolate.Diagnostics" />
+        <PackageReference Include="HotChocolate.Data.EntityFramework" />
     </ItemGroup>
 
 </Project>
diff --git a/src/Defaults/SourceSchemaDefaults/Extensions.cs b/src/Defaults/SourceSchemaDefaults/Extensions.cs
index 3d08fea..8f63a94 100644
--- a/src/Defaults/SourceSchemaDefaults/Extensions.cs
+++ b/src/Defaults/SourceSchemaDefaults/Extensions.cs
@@ -1,4 +1,5 @@
 using HotChocolate.Execution.Configuration;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Extensions.Hosting;
@@ -7,7 +8,8 @@ public static class Extensions
 {
     public static IRequestExecutorBuilder AddDefaultSettings(
         this IRequestExecutorBuilder builder,
-        bool registerNodeInterface = true)
+        bool registerNodeInterface = true
+    )
     {
         builder.AddGlobalObjectIdentification(registerNodeInterface);
         builder.AddInstrumentation();
@@ -16,6 +18,18 @@ public static class Extensions
         builder.AddQueryContext();
         builder.ModifyCostOptions(x => x.EnforceCostLimits = false);
         builder.ExportSchemaOnStartup();
+        builder.AddNitro(
+            (sp, c) =>
+            {
+                c.ApiId = sp.GetRequiredService<IConfiguration>()["Nitro:ApiId"]!;
+                c.ApiKey = sp.GetRequiredService<IConfiguration>()["Nitro:ApiKey"]!;
+                c.Stage = "dev";
+                c.Metrics.Enabled = true;
+                c.ServerUrl = "https://api.chillicream.cloud";
+                c.TelemetryUrl = "https://otel.chillicream.cloud";
+                c.EnableOperationReporting = false;
+            }
+        );
         return builder;
     }
 }
diff --git a/src/Gateway/Demo.Gateway.csproj b/src/Gateway/Demo.Gateway.csproj
index d0428a4..52ad4f1 100644
--- a/src/Gateway/Demo.Gateway.csproj
+++ b/src/Gateway/Demo.Gateway.csproj
@@ -11,6 +11,7 @@
 
     <ItemGroup>
         <PackageReference Include="HotChocolate.Fusion.AspNetCore"/>
+        <PackageReference Include="ChilliCream.Nitro.Fusion"/>
         <PackageReference Include="Microsoft.AspNetCore.HeaderPropagation"/>
     </ItemGroup>
 
diff --git a/src/Gateway/Program.cs b/src/Gateway/Program.cs
index cbaedcd..c2081cf 100644
--- a/src/Gateway/Program.cs
+++ b/src/Gateway/Program.cs
@@ -1,37 +1,51 @@
 using HotChocolate;
 using HotChocolate.AspNetCore;
 using HotChocolate.Fusion.AspNetCore;
+using OpenTelemetry.Log;
+using OpenTelemetry.Trace;
 
 var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults(Env.GatewayApi, Env.Version);
 
-builder.Services
-    .AddCors()
+builder
+    .Services.AddCors()
     .AddHeaderPropagation(c =>
     {
         c.Headers.Add("GraphQL-Preflight");
         c.Headers.Add("Authorization");
     });
 
-builder.Services
-    .AddHttpClient("Fusion")
-    .AddHeaderPropagation();
+builder.Services.AddHttpClient("Fusion").AddHeaderPropagation();
 
 builder
     .AddGraphQLGateway()
-    .AddFileSystemConfiguration("./gateway.far")
+    .AddNitro(x =>
+    {
+        x.ApiId = "QXBpCmcwMTk4ZWRlZWM0YmQ3OTY5ODc2MTA4NDU3M2NhMzdmNQ==";
+        x.Stage = "dev";
+        x.ApiKey = "WMqJUtANoKre4h6HyAk4rfUk5vm05pYvjLCWhDQ5ds59n09WlET1736kxpWa91cF";
+        x.Metrics.Enabled = true;
+        x.ServerUrl = "https://api.chillicream.cloud";
+        x.TelemetryUrl = "https://otel.chillicream.cloud";
+    })
     .ModifyRequestOptions(o => o.CollectOperationPlanTelemetry = true)
-    .AddDiagnosticEventListener(sp =>
-        new DebugDiagnosticListener(
-            sp.GetRequiredService<IRootServiceProviderAccessor>()
-                .ServiceProvider
-                .GetRequiredService<ILoggerFactory>()));
+    .AddDiagnosticEventListener(sp => new DebugDiagnosticListener(
+        sp.GetRequiredService<IRootServiceProviderAccessor>()
+            .ServiceProvider.GetRequiredService<ILoggerFactory>()
+    ));
+
+builder
+    .Services.AddOpenTelemetry()
+    .WithLogging(x => x.AddNitroExporter())
+    .WithTracing(x => x.AddNitroExporter());
 
 var app = builder.Build();
 
 app.UseCors(c => c.AllowAnyHeader().AllowAnyMethod().AllowAnyOrigin());
 app.UseHeaderPropagation();
-app.MapGraphQL().WithOptions(new GraphQLServerOptions { Tool = {  ServeMode = GraphQLToolServeMode.Insider } });
+app.MapGraphQL()
+    .WithOptions(new GraphQLServerOptions { Tool = { ServeMode = GraphQLToolServeMode.Insider } });
 
+Console.WriteLine("Gateway API is running...");
 app.Run();
diff --git a/src/SourceSchemas/Accounts/appsettings.json b/src/SourceSchemas/Accounts/appsettings.json
index 4d56694..f976b45 100644
--- a/src/SourceSchemas/Accounts/appsettings.json
+++ b/src/SourceSchemas/Accounts/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZWZmZTU1MWY3YTQwODkxN2Y3ZDlkNWNmMjZiNw==",
+    "ApiKey": "hYed0Qq2b5kGqm35I3aIK05wn0HHEaVIK4JsCYURefAXNQ25f291bx4TE3AfyU93"
+  },
   "AllowedHosts": "*"
 }
diff --git a/src/SourceSchemas/Inventory/appsettings.json b/src/SourceSchemas/Inventory/appsettings.json
index 4d56694..86c2fda 100644
--- a/src/SourceSchemas/Inventory/appsettings.json
+++ b/src/SourceSchemas/Inventory/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZjAwMmRkMjQ3ZjRlYWQ0MDI1MTVkMDA5NzRiMA==",
+    "ApiKey": "k26ivPQNsgrGFz2Fx1S3E4zmgz7hnXGxLfCBRH3HPQ5CRYSKO34m10266QhIH27l"
+  },
   "AllowedHosts": "*"
 }
diff --git a/src/SourceSchemas/Order/appsettings.json b/src/SourceSchemas/Order/appsettings.json
index 4d56694..6423556 100644
--- a/src/SourceSchemas/Order/appsettings.json
+++ b/src/SourceSchemas/Order/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZjAwM2MwYmQ3MzZhOTdjODAzN2U2ZmUwMDE5Nw==",
+    "ApiKey": "l1mHDvnPTzAQv5OW1gYH8w0gHeeGY0fIPMlYorTvs0qBXazpcCwCHzkChiDABfjC"
+  },
   "AllowedHosts": "*"
 }
diff --git a/src/SourceSchemas/Payments/appsettings.json b/src/SourceSchemas/Payments/appsettings.json
index 4d56694..39cce9e 100644
--- a/src/SourceSchemas/Payments/appsettings.json
+++ b/src/SourceSchemas/Payments/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZjAwNGU5NDk3ZWFkOTgxNDBiODJlYWZkNzg1OQ==",
+    "ApiKey": "wH3njNMaWDETXjdzGQyBQjs4ADHRart6Z6My0TVGJ1FpJUPVvDNWyGQqp0K0UMWw"
+  },
   "AllowedHosts": "*"
 }
diff --git a/src/SourceSchemas/Products/appsettings.json b/src/SourceSchemas/Products/appsettings.json
index 4d56694..e8db1f7 100644
--- a/src/SourceSchemas/Products/appsettings.json
+++ b/src/SourceSchemas/Products/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZjAwNTgxOTI3YjFkYWE0OTJhYjhjNWY2NTkwNQ==",
+    "ApiKey": "6CsA9GOZTrrDfH8fIvq4loPjykVPwTU9Ui80OurL6vrS6OfniGA6LcPLmLhLbAmd"
+  },
   "AllowedHosts": "*"
 }
diff --git a/src/SourceSchemas/Reviews/appsettings.json b/src/SourceSchemas/Reviews/appsettings.json
index 4d56694..f09c469 100644
--- a/src/SourceSchemas/Reviews/appsettings.json
+++ b/src/SourceSchemas/Reviews/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZjAwNjk5MWQ3NTU5OTI0MzdlYzM3OTg2YjMzNA==",
+    "ApiKey": "fo1l2R4N2LgA3X6UrLaRHFTZrPDOxN3aXv1qLFs2z41PGyqD5nJxx9lLHAg3jG53"
+  },
   "AllowedHosts": "*"
 }
diff --git a/src/SourceSchemas/Shipping/appsettings.json b/src/SourceSchemas/Shipping/appsettings.json
index 4d56694..2198286 100644
--- a/src/SourceSchemas/Shipping/appsettings.json
+++ b/src/SourceSchemas/Shipping/appsettings.json
@@ -5,5 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "Nitro": {
+    "ApiId": "QXBpCmcwMTk4ZjAwODE5YWU3ZjMyYTAzMmM1YzkzNmI5NWNmOA==",
+    "ApiKey": "btGGUfH4oFormpa9TAxlxzQDqFH5hF4pukCDZIL2vfUX0kY0sIYlij21vgFuIPtY"
+  },
   "AllowedHosts": "*"
 }
